bakerydemo/recipes/blocks.py

```diff
@@ -0,0 +1,77 @@
+from django import forms
+from wagtail.contrib.table_block.blocks import TableBlock
+from wagtail.contrib.typed_table_block.blocks import TypedTableBlock
+from wagtail.core.blocks import (
+    CharBlock,
+    ChoiceBlock,
+    FloatBlock,
+    ListBlock,
+    PageChooserBlock,
+    RichTextBlock,
+    StreamBlock,
+    StructBlock,
+    TextBlock,
+)
+from wagtail.documents.blocks import DocumentChooserBlock
+from wagtail.embeds.blocks import EmbedBlock
+from wagtail.images.blocks import ImageChooserBlock
+
+from bakerydemo.base.blocks import BlockQuote, HeadingBlock, ImageBlock
+
+
+class RecipeStepBlock(StructBlock):
+    text = RichTextBlock(features=["bold", "italic", "link"])
+    difficulty = ChoiceBlock(
+        widget=forms.RadioSelect,
+        choices=[("S", "Small"), ("M", "Medium"), ("L", "Large")],
+        default="S",
+    )
+
+    class Meta:
+        template = 'blocks/recipe_step_block.html'
+        icon = 'tick'
+
+
+class RecipeStreamBlock(StreamBlock):
```

Comment:
Additions compared to existing BaseStreamBlock:

- Groups
- TableBlock
- TypedTableBlock
- ListBlock
- `min_num` and `max_num` validation